proposals/0000-specify-dump-json.md

```diff
@@ -0,0 +1,149 @@
+ - `"message"` corresponding to `DecoratedSDoc` of `diagnosticMessage`
+ - `"reason"` corresponding to `diagnosticReason`
+ 
+ The top level version is crucial for indicating to downstream consumers which JSON schema they must comply with in the case that the schema is updated in the future (which is quite likely).
+ 
+For demonstrative purposes, here is an example valid instance of the schema.
+```json
+{
+  "version": "1.0.0",
+  "diagnostics": [
+    {
+      "span": {
+        "file": "typecheck/should_fail/T2414.hs",
+        "startLine": 9,
+        "startCol": 13,
+        "endLine": 9,
+        "endCol": 17
+      },
+      "severity": "Error",
+      "code": 27958,
+      "message": "    • Couldn't match type ‘b0’ with ‘(Bool, b0)’ \n Expected: b0 -> Maybe (Bool, b0) \nActual: b0 -> Maybe b0 \n• In the first argument of ‘unfoldr’, namely ‘Just’ \nIn the expression: unfoldr Just \nIn an equation for ‘f’: f = unfoldr Just",
+      "warnReason": {
+        "reason": "ErrorWithoutFlag"
+      }
+    }
+  ]
+}
+```
+
+The schema itself will be brought into version control of the GHC repo and tests will be added to ensure that GHC-dump complies with the JSON schema. This will ensure that changes to the diagnostics will result in appropriate changes to the schema. In addition, a wiki page will be created to fill in the gaps for anyone needing to understand the project. Otherwise, I will be available to perform maintenance. 
+
+One of the major benefits of utilizing a JSON schema is that the expected JSON payload can be well-defined for consumers of the messages. This has massive benefits, as consumers can presume the structure of the output without having to analyze the contents for the presence or absence of particular bits of data. However, one drawback may be the over specification of the output. There may be some opportunities in which flexibility is a benefit for the output. However, this schema can be adapted as further feedback rolls in (with incremented version), making the good net outweigh the bad.
+
+In addition to adding a `-dump-json` flag, it may also prove useful to provide a `-dump-json-schema` flag which simply produces the relevant JSON schema for that particular version of GHC. This I leave open for discussion. Provided that the schema is in an easy to find location, it may be overkill.
+
+The schmema evolution process is currently undetermined, though I imagine that due to the infrequency with which the schema will need to be changed, it can be handled on a case-by-base basis. Though keeping a running list of all relevant stakeholders that may need to be informed could be a good idea. 
```

Comment:
```suggestion
The schema evolution process is currently undetermined, though I imagine that due to the infrequency with which the schema will need to be changed, it can be handled on a case-by-base basis. Though keeping a running list of all relevant stakeholders that may need to be informed could be a good idea. 
```